### PR TITLE
Fix tap

### PR DESCRIPTION
**Pull request by JingLin0**
# Description of change
(write a short description or paste a link to JIRA)

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by przywartya — APPROVED** [review]

File: tap_pipedrive/streams/activity_types.py [hunk 1]
```diff
@@ -7,4 +7,4 @@ class ActivityTypesStream(PipedriveStream):
     key_properties = [
         "id",
     ]
-    state_field = "update_time"
+    state_field = "add_time"
```
Comment (paddie) [inline tap_pipedrive/streams/activity_types.py]: use `add_time` only if the `updated_at` does not have a value.

File: tap_pipedrive/streams/recents/dynamic_typing/__init__.py [hunk 2]
```diff
@@ -14,59 +14,5 @@ class DynamicTypingRecentsStream(RecentsStream):
...
-                        # be marked mandatory for some amount of time and not
-                        # mandatory for another amount of time
-                        property_content["type"].append("null")
-
-                        schema["properties"][property["key"]] = property_content
-
-            except Exception as e:
-                raise e
-
             self.schema_cache = schema
```
Comment (paddie) [inline tap_pipedrive/streams/recents/dynamic_typing/__init__.py:17]: 👏

**Review by paddie — CHANGES_REQUESTED** [review]

File: tap_pipedrive/streams/filters.py [hunk 3]
```diff
@@ -7,4 +7,4 @@ class FiltersStream(PipedriveStream):
     key_properties = [
         "id",
     ]
-    state_field = "add_time"
+    state_field = "update_time"
```
Comment (paddie) [inline tap_pipedrive/streams/filters.py:10]: As I understood it, the `update_time` does not have values (potentially until they get updated). My previous comment was to fall back to `add_time` if the `update_time` does not have a value to guarantee that you always have a value.

**Review by paddie — CHANGES_REQUESTED** [review]

File: tap_pipedrive/streams/filters.py (hunk 3 above)
Comment (JingLin0) [reply tap_pipedrive/streams/filters.py:10]: It turns out if update_time is None, the state will be the start_date from config file. So we do not need to change update_time to add_time

**Review by paddie — APPROVED** [review]

**Pull request merged**